File: app/storage/file/remote_skill_source_store.py
"""远端 Skill 来源配置的文件存储。

布局：data/remote_skill_sources/{source_name}.json
"""

from __future__ import annotations

from pathlib import Path
from typing import Any

from app.config.settings import get_settings
from app.storage.file.base import list_json_ids, read_json, write_json_atomic


class RemoteSkillSourceStore:
    """将 RemoteSkillSourceConfig 序列化为 data/remote_skill_sources/{name}.json。"""

    def _path(self, source_name: str) -> Path:
        safe = source_name.replace("/", "_").replace("\\", "_")
        return get_settings().data_dir / "remote_skill_sources" / f"{safe}.json"

    def save(self, data: dict[str, Any]) -> None:
        write_json_atomic(self._path(data["source_name"]), data)

    def get(self, source_name: str) -> dict[str, Any] | None:
        return read_json(self._path(source_name))

    def delete(self, source_name: str) -> bool:
        path = self._path(source_name)
        if path.exists():
            path.unlink()
            return True
        return False

    def list_all(self) -> list[dict[str, Any]]:
        ids = list_json_ids(get_settings().data_dir / "remote_skill_sources")
        result = []
        for name in ids:
            data = self.get(name)
            if data:
                result.append(data)
        return result

File: miniagents-assembled/src/main/java/com/codex/miniagents/infrastructure/storage/file/RemoteSkillSourceStore.java
package com.codex.miniagents.infrastructure.storage.file;

import com.codex.miniagents.config.MiniAgentsProperties;
import com.codex.miniagents.skills.model.RemoteSkillSourceConfig;

import org.springframework.stereotype.Repository;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Repository
public class RemoteSkillSourceStore {
    private final Path baseDir;

    private final FileJsonSupport jsonSupport;

    public RemoteSkillSourceStore(MiniAgentsProperties properties, FileJsonSupport jsonSupport) {
        this.baseDir = properties.getDataDir().resolve("remote_skill_sources");
        this.jsonSupport = jsonSupport;
    }

    private Path path(String sourceName) {
        String safe = sourceName.replace("/", "_").replace("\\", "_");
        return baseDir.resolve(safe + ".json");
    }

    public void save(RemoteSkillSourceConfig config) {
        jsonSupport.writeJsonAtomic(path(config.getSourceName()), config);
    }

    public Optional<RemoteSkillSourceConfig> get(String sourceName) {
        return jsonSupport.readJson(path(sourceName), RemoteSkillSourceConfig.class);
    }

    public boolean delete(String sourceName) {
        Path p = path(sourceName);
        try {
            return Files.deleteIfExists(p);
        } catch (IOException e) {
            throw new RuntimeException("Failed to delete remote skill source: " + sourceName, e);
        }
    }

    public List<RemoteSkillSourceConfig> listAll() {
        List<RemoteSkillSourceConfig> results = new ArrayList<>();
        for (String sourceName : jsonSupport.listJsonIds(baseDir)) {
            get(sourceName).ifPresent(results::add);
        }
        return results;
    }
}
